src/shared/sol-util.c

```diff
@@ -312,14 +315,37 @@ sol_util_str_split(const struct sol_str_slice slice, const char *delim, size_t m
 static int
 get_progname(char *out, size_t size)
 {
-    char cwd[PATH_MAX] = { NULL }, *execfn;
+    char cwd[PATH_MAX] = { NULL }, *execfn = NULL;
 
     if (getcwd(cwd, sizeof(cwd)) == NULL)
-        return -1;
+        return -errno;
 
+#ifdef HAVE_SYS_AUXV_H
     execfn = (char *)getauxval(AT_EXECFN);
```

Comment:
I guess getauxval() is not defined only on !Linux, right? If that's the case create a sol-util-linux.[ch] and define it there, do not use it in RIOT and other systems where getting the program name would not have a meaning.

Actually, the only user is sol_util_get_rootdir() that has no meaning outside of linux as Contiki, RIOT and others won't have rootdir() or a filesystem at all.